File: app/src/main/java/com/morihacky/android/rxjava/rxbus/RxBusDemo_TopFragment.java
package com.morihacky.android.rxjava.rxbus;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import butterknife.ButterKnife;
import butterknife.OnClick;
import com.morihacky.android.rxjava.MainActivity;
import com.morihacky.android.rxjava.R;
import com.morihacky.android.rxjava.fragments.BaseFragment;

public class RxBusDemo_TopFragment extends BaseFragment {

  private RxBus _rxBus;

  @Override
  public View onCreateView(
      LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
    View layout = inflater.inflate(R.layout.fragment_rxbus_top, container, false);
    ButterKnife.bind(this, layout);
    return layout;
  }

  @Override
  public void onActivityCreated(@Nullable Bundle savedInstanceState) {
    super.onActivityCreated(savedInstanceState);
    _rxBus = ((MainActivity) getActivity()).getRxBusSingleton();
  }

  @OnClick(R.id.btn_demo_rxbus_tap)
  public void onTapButtonClicked() {
    if (_rxBus.hasObservers()) {
      _rxBus.send(new RxBusDemoFragment.TapEvent());
    }
  }
}

File: app/src/main/java/com/morihacky/android/rxjava/rxbus/RxBusDemo_TopFragment.kt
package com.morihacky.android.rxjava.rxbus

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import com.morihacky.android.rxjava.MainActivity
import com.morihacky.android.rxjava.R
import com.morihacky.android.rxjava.fragments.BaseFragment
import kotlinx.android.synthetic.main.fragment_rxbus_top.*

class RxBusDemo_TopFragment : BaseFragment() {

    private lateinit var rxBus: RxBus

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ) = inflater.inflate(R.layout.fragment_rxbus_top, container, false)

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        rxBus = (activity as MainActivity).rxBusSingleton
        btn_demo_rxbus_tap.setOnClickListener {
            if (rxBus.hasObservers()) {
                rxBus.send(RxBusDemoFragment.TapEvent())
            }
        }
    }
}
